sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -191,7 +193,12 @@ private[hive] class HiveClientImpl(
   /** Returns the configuration for the current session. */
   def conf: HiveConf = state.getConf
 
-  private val userName = conf.getUser
+  private val userName: String = try {
+    val ugi = HiveUtils.getUGI
+    ugi.getShortUserName
```

Comment:
How do we prove this is the right fix? What if there is no UGI?

Reply:
It will return system user information if there is no ugi. Actually this source code of `conf.getUser `is:

    ```
public String getUser() throws IOException {
        try {
             UserGroupInformation ugi = Utils.getUGI(); 
             return ugi.getUserName();
         } catch (LoginException var2) { 
             throw new IOException(var2); 
         } 
     }
```
this chang just use `ugi.getShortUserName` instead of `ugi.getUserName()`

Reply:
Looks like `Utils` is accessible from here, so if we just want to leverage Hive code and call `getShortUserName` instead, we can just do it with one-liner.

```
private val userName = org.apache.hadoop.hive.shims.Utils.getUGI.getShortUserName
```

Reply:
> org.apache.hadoop.hive.shims.Utils.getUGI.getShortUserName

Yep, but `shims.Utils` is not compatible with hive0.*. This file is added after hive1.*, so, copy the source code here.

Reply:
Ah OK. Didn't recognize it doesn't match with some versions. Makes sense to me.

Reply:
f1eb8cfe9c32e74822b7701588d6e961d0d767ea was for this, sorry I had to track down commits as well.